config/src/test/java/org/springframework/security/config/annotation/web/configurers/oauth2/server/resource/OAuth2ResourceServerConfigurerTests.java

```diff
@@ -526,6 +528,107 @@ public void requestWhenSessionManagementConfiguredThenUserConfigurationOverrides
 		assertThat(result.getRequest().getSession(false)).isNotNull();
 	}
 
+	// -- custom bearer token resolver
+
+	@Test
+	public void requestWhenBearerTokenResolverAllowsRequestBodyThenEitherHeaderOrRequestBodyIsAccepted()
```

Comment:
Please add a test where the token is in the header and body AND another test where the token is in header and query param

Reply:
We have unit tests for these. For my education, how do you gauge what use cases need both an integration test as well as a unit test?

Where my mind goes right now is these integration tests confirm that things got wired correctly and behave accordingly, and unit tests confirm that, given XYZ arbitrary wirings, the contract is correctly upheld.

I don't have a problem adding more tests, I just want to learn from you how you prioritize your own efforts with things like this.

Reply:
Also, this specific recommendation actually uncovered a bug (awesome!) which the unit test would not have uncovered, which is why I'm interested in your thinking around it.

Reply:
Sorry I missed that - the unit tests cover the scenarios I suggested. Those tests should be in the units tests.

> Where my mind goes right now is these integration tests confirm that things got wired correctly and behave accordingly

Agreed, the integration tests should confirm things got wired correctly and behave as expected.